### PR TITLE
fix: potential circular dependency

### PR DESCRIPTION
**Pull request by ntn-x2**
The lookup pallet introduced a dependency on the `runtime-common` crate, which should not really be there. While working on DIP, where the reverse relationship is introduced, we have a circular dependency.

This PR abstracts in the lookup pallet over the definition of `Balance` by re-using the trait constraints that the `pallet-balances` defines. It also resolves one lint warning regarding an unused function.

### CODE REVIEW

**Review by weichweich — COMMENTED** [review]
This shouldn't have happened. Good catch

File: pallets/pallet-did-lookup/src/lib.rs [hunk 1]
```diff
@@ -107,8 +109,19 @@ pub mod pallet {
...
+			+ Member
+			+ AtLeast32BitUnsigned
+			+ Codec
+			+ Default
+			+ Copy
+			+ MaybeSerializeDeserialize
+			+ sp_std::fmt::Debug
+			+ MaxEncodedLen
+			+ TypeInfo
+			+ FixedPointOperand;
```
Comment (weichweich) [inline pallets/pallet-did-lookup/src/lib.rs:122]: Can you not simply use MutateHold::Balance? Do you need the additional trait bounds?

Comment (ntn-x2) [reply pallets/pallet-did-lookup/src/lib.rs]: I was able to bring it down to [this](https://github.com/KILTprotocol/kilt-node/pull/547/commits/f57c084f26949ca0d95dc76b5c97d89f65e88700). If I remove the trait bounds completely, everything becomes red.

Comment (ntn-x2) [reply pallets/pallet-did-lookup/src/lib.rs]: Ok maybe this works... https://github.com/KILTprotocol/kilt-node/pull/547/commits/609a41b18776d2fc0304823ce894bd783afaccd0

**Review by weichweich — APPROVED** [review]

**Review by Ad96el — APPROVED** [review]

**Comment by Ad96el** [comment]
Closed by accident :D

**Pull request merged**